javascript/memoize/memoize.ts

```diff
@@ -0,0 +1,20 @@
+type MemoizeFn<T> = (...args: any[]) => T;
```

Comment:
P4: 입력 타입에도 제네릭을 사용할 수 있게 바꿀 수 있을 것 같아요.

```suggestion
type MemoizeFn<T, Args extends any[]> = (...args: Args) => T;
```

Reply:
72fae40bf2848ec71f234dd2fc91e930f819e2a3 타입 수정해봤습니다!